go-controller/pkg/ovn/egressip.go

```diff
@@ -611,15 +619,27 @@ func (oc *Controller) addEgressNode(egressNode *kapi.Node) error {
 
 func (oc *Controller) deleteEgressNode(egressNode *kapi.Node) error {
 	klog.V(5).Infof("Egress node: %s about to be removed", egressNode.Name)
-	if stdout, stderr, err := util.RunOVNNbctl(
-		"remove",
-		"logical_switch_port",
-		types.EXTSwitchToGWRouterPrefix+types.GWRouterPrefix+egressNode.Name,
-		"options",
-		"nat-addresses=router",
-	); err != nil {
-		klog.Errorf("Unable to remove GARP configuration on external logical switch port for egress node: %s, stdout: %s, stderr: %s, err: %v", egressNode.Name, stdout, stderr, err)
+	// TODO: migrate this code to the operations model once there is a good way to delete items from maps
+	lsp := &nbdb.LogicalSwitchPort{
+		Name: types.EXTSwitchToGWRouterPrefix + types.GWRouterPrefix + egressNode.Name,
 	}
+	if err := oc.nbClient.Get(lsp); err != nil {
+		klog.Errorf("Unable to remove GARP configuration on external logical switch port for egress node: %s, err: %v", egressNode.Name, err)
+	} else {
+		ops, err := oc.nbClient.Where(lsp).Mutate(lsp, model.Mutation{
+			Field:   &lsp.Options,
+			Mutator: ovsdb.MutateOperationDelete,
+			Value:   []string{"nat-addresses"},
+		})
+		if err != nil {
+			klog.Errorf("Unable to remove GARP configuration on external logical switch port for egress node: %s, err: %v", egressNode.Name, err)
+		}
+		_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
+		if err != nil {
+			klog.Errorf("Unable to remove GARP configuration on external logical switch port for egress node: %s, err: %v", egressNode.Name, err)
```

Comment:
NIT: It would good to make this log message a little different than the message on line 635, so by looking at the code we can more easily tell which one of the two happened.

Reply:
done